File: TodoListBackend/TodoList.Domain/Entities/TodoItem.cs
namespace TodoList.Domain.Entities;

public class TodoItem : Base.Entity
{
    public string Title { get; set; }
    public string Description { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? CompletedAt { get; set; }

    public TodoItem(string title, string description) 
    {
        Title = title;
        Description = description;
        CreatedAt = DateTime.Now;
    }

    public void Update(string title, string description, bool isCompleted)
    {
        Title = title;
        Description = description;
        IsCompleted = isCompleted;
        CompletedAt = isCompleted ? DateTime.Now : null;
    }

    public void Complete() 
    {
        IsCompleted = true;
        CompletedAt = DateTime.Now; 
    }
}

File: TodoListApp/app/src/main/java/com/todolist/data/model/TodoItem.kt
package com.todolist.data.model

import com.google.gson.annotations.SerializedName

data class TodoItem(
    @SerializedName("id") var id: String? = null,
    @SerializedName("title") var title: String,
    @SerializedName("description") var description: String? = null,
    @SerializedName("isCompleted") var isCompleted: Boolean = false,
    var createdAt: String = ""
) {
    constructor(title: String) : this(
        title = title,
        description = "",
        isCompleted = false,
        createdAt = ""
    )
}
